include/zephyr/drivers/video-controls.h

```diff
@@ -31,42 +38,91 @@ extern "C" {
 
 /**
  * @name Control classes
+ *
+ * List of control classes, to group related controls together in blocks.
+ *
+ * This list is kept identical to the Linux kernel definitions.
+ *
  * @{
  */
-#define VIDEO_CTRL_CLASS_GENERIC	0x00000000	/**< Generic class controls */
-#define VIDEO_CTRL_CLASS_CAMERA		0x00010000	/**< Camera class controls */
-#define VIDEO_CTRL_CLASS_MPEG		0x00020000	/**< MPEG-compression controls */
-#define VIDEO_CTRL_CLASS_JPEG		0x00030000	/**< JPEG-compression controls */
-#define VIDEO_CTRL_CLASS_VENDOR		0xFFFF0000	/**< Vendor-specific class controls */
+#define VIDEO_CTRL_CLASS_BASE			0x00980900 /**< Base class controls */
+#define VIDEO_CTRL_CLASS_CODEC			0x00990900 /**< Stateful codec controls */
+#define VIDEO_CTRL_CLASS_CAMERA			0x009a0900 /**< Camera class controls */
+#define VIDEO_CTRL_CLASS_FM_TX			0x009b0900 /**< FM Modulator controls */
+#define VIDEO_CTRL_CLASS_FLASH			0x009c0900 /**< Camera flash controls */
+#define VIDEO_CTRL_CLASS_JPEG			0x009d0900 /**< JPEG-compression controls */
+#define VIDEO_CTRL_CLASS_IMAGE_SOURCE		0x009e0900 /**< Image source controls */
+#define VIDEO_CTRL_CLASS_IMAGE_PROC		0x009f0900 /**< Image processing controls */
```

Comment:
I think we should keep the same name as Linux, otherwise, it will cause more confusion. In Linux:

`#define V4L2_CTRL_CLASS_IMAGE_PROC 0x009f0000`
`#define V4L2_CID_IMAGE_PROC_CLASS_BASE (V4L2_CTRL_CLASS_IMAGE_PROC | 0x900)`

Reply:
I was unsure what was the best: reducing the complexity, or keep all the remapping as Linux has even if it carries no information in the context of Zephyr.
I will implement as you said, thank you for the feedback!

Reply:
No, I didn't say we need to keep all the remapping as Linux. Just here, we should do :

`#define VIDEO_CTRL_CLASS_IMAGE_PROC_**BASE**		0x009f0900`

no need to define V4L2_CTRL_CLASS_IMAGE_PROC though.

Because when looking at what is currently done:

`#define VIDEO_CTRL_CLASS_IMAGE_PROC		0x009f0900`

people will wonder why it is different in Linux where this macro is **0x009f0000**

Reply:
Thank you I missed that point! This is now updated through the commits.
One edge-case about the BASE class here though: https://github.com/zephyrproject-rtos/zephyr/pull/78802#discussion_r1783121678